### PR TITLE
Process message

### LINKED ISSUE #4906: v8 processing message from parent window
The branch `next` is for the development of v8 using mobx. It lacks of message processing support that is required by this [ticket](https://github.com/TerriaJS/nationalmap/issues/981).

It should be able to process the message similar to the following from a parent window and display the item in the map.
```
{
  "catalog": [
    {
      "name": "Credit Licence Dataset - National Map",
      "type": "magda",
      "recordId": "dist-dga-548075a1-c36e-4837-bf26-cc00567b5c23",
      "url": "https://dev.magda.io/"
    }
  ],
  "baseMapName": "Positron (Light)"
}
```
The above message can be sent from `magda-server-client`. For details, see [this tickiet](https://github.com/TerriaJS/nationalmap/issues/981).

This [issue](https://github.com/TerriaJS/terriajs/issues/4719) might be related.

### PR DESCRIPTION
**Pull request by mwu2018**
### What this PR does

Fixes #4906 

The interpretStartData function should wrap push operation in runInAction.

### Checklist

-   [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

***Technical note for developers***
A `TerriaMap` with [PR](https://github.com/TerriaJS/TerriaMap/issues/494) using `terriajs` with this PR will be able to process message containing magda item and post by its parent window. However this PR only adds the magda item to the workbench but does not automatically show it on the map. The root cause of this behaviour is likely due to racing conditions, which needs to be addressed separately. To demonstrate this, follow the steps below.

1. Check out [this branch](https://github.com/TerriaJS/TerriaMap/tree/enable-post-message) and
  ```
  yarn install
  yarn build
  yarn start
  ```
2. Check out [this branch](https://github.com/magda-io/magda/tree/3019-test-do-not-merge) and
    ```
    cd magda-web-client
    yarn install
    NODE_OPTIONS=--max_old_space_size=4096 yarn run gulp build
    yarn start
    ```
    After a while, [this page](http://localhost:6108) will be opened automatically. 
    Copy and paste `http://localhost:6108/dataset/ds-dga-fa0b0d71-b8b8-4af8-bc59-0b000ce0d5e4/details?q=credit%20license` to the address bar -> Click on `Preview Anyway` in `Map Preview` -> Click on `Open in NationalMap`. 
3. The following page will be opened: 
   image
   
   Note that the magda item appears in the workbench but not on the map. To make it show up on the map, do either of the followings:
    - Click on `ABOUT DATA` button -> Click on `Done`.
    - Click on `Explore map data` -> Scroll down to the bottom of `Data Catalogue` -> Click on `Credit Licence Dataset - National Map` -> Click on `Done`.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -3,8 +3,11 @@ Change Log
 
 ### MobX Development
 
-#### next release (8.0.0-alpha.59)
+#### next release (8.0.0-alpha.60)
```
Comment (soyarsauce) [inline CHANGES.md]: Same as other PR: We don't need this, it's a bit confusing - the next release is .59

**Review by soyarsauce — CHANGES_REQUESTED** [review]

File: CHANGES.md (hunk 1 above)
Comment (mwu2018) [reply CHANGES.md]: fixed

**Comment by mwu2018** [comment]
@soyarsauce Are you happy to merge this PR?

**Review by soyarsauce — APPROVED** [review]

**Comment by soyarsauce** [comment]
BTW @mwu2018 since you are using vscode, you may find this extension handy to help enforce prettier at the client level rather than waiting for ci! 🙂 
https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

**Pull request merged**